app/models/collection_item.rb

```diff
@@ -320,4 +320,33 @@ def notify_of_reveal
       end
     end
   end
+
+  after_update :notify_of_unrevealed_or_anonymous
+  def notify_of_unrevealed_or_anonymous
+    # This CollectionItem's anonymous/unrevealed status can only affect the
+    # item's status if (a) the CollectionItem is approved by the user and (b)
+    # the item is a work. (Bookmarks can't be anonymous/unrevealed at the
+    # moment.)
+    return unless approved_by_user? && item.is_a?(Work)
+
+    # Check whether anonymous/unrevealed is becoming true, when the work
+    # currently has it set to false:
+    newly_anonymous = (saved_change_to_anonymous?(to: true) && !item.anonymous?)
+    newly_unrevealed = (saved_change_to_unrevealed?(to: true) && !item.unrevealed?)
+
+    return unless newly_unrevealed || newly_anonymous
+
+    # Don't notify if it's one of the work creators who is changing the work's
+    # status.
+    return if item.users.include?(User.current_user)
+
+    item.users.each do |user|
+      next if user.preference.collection_emails_off
```

Comment:
I have a hunch we might want to ignore the preference for something as important as this, but I'm going to check.

Reply:
Yup, Support would rather deal with users grumpy about an email than grumpy about their work disappearing. If you could remove this check and update the text of the help file with the new version Docs provided, this should be ready to merge. 🎉 

The old help text line: https://github.com/otwcode/otwarchive/blob/57e7cab589abc7cc666ddf4d3abe768d801b3493/public/help/collection-preferences.html#L9

The new version:
```html
  <dd>Enable this option if you would prefer not to receive email alerts from collections, such as creator and work reveals. We’ll still email you if your username or work is hidden after being added to a collection. Notifications will still be delivered to your Archive inbox unless you have chosen to disable them.</dd>
```

Reply:
About the wording: Users currently aren't notified when their works are revealed. (a) Should there be an issue to add those notifications? (b) Is it okay that the help text is more about what it's intended to control than about what it currently controls? (Collection invitations, mostly.)

Reply:
I'll be honest: I don't think we put that much thought into the original version of the help text or even verified that it does as it says. There's no paper trail beyond a decision to add some hypothetical text in a November 2011 meeting, the issue itself ([AO3-2704](https://otwarchive.atlassian.net/browse/AO3-2704)), and Ariana's pull request in 2012 (#662). So I wouldn't really be inclined to just make an issue about that.

Let me look into what the preference _actually_ does and then ping Docs to see if we can get wording that reflects the present reality.

Reply:
All righty, here we go:

>Enable this option if you would prefer not to receive email alerts when your works are added or invited to collections.

Reply:
Thanks for checking with Docs to figure out what the wording should be!